spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -855,19 +855,57 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           None
 
         case EqualTo(left, right) =>
-          val leftExpr = exprToProtoInternal(left, inputs)
-          val rightExpr = exprToProtoInternal(right, inputs)
+          val zero = Literal.default(left.dataType)
+          val negZero = UnaryMinus(zero)
```

Comment:
RIght type may be different from left type. So we need Right version of `zero` and `negZero`?

For `negZero` you can do `Literal(-0.0)` and `Literal(-0.0f)` instead of UnaryMinus

Reply:
resolved.